flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -349,27 +351,29 @@ class JobManager(
 
               case JobStatus.CANCELED =>
                 jobInfo.client ! decorateMessage(
-                  Failure(
-                    new JobCancellationException(
-                      jobID,
-                    "Job was cancelled.", error)
+                  Failure(new SerializedThrowable(
+                      new JobCancellationException(
```

Comment:
Why do we have to serialize the `JobCancellationException` here? Isn't it enough to only serialize the error transferred with the `JobStatusChanged` message?